### PR TITLE
feat(accordion-item,stepper-item,block,panel): add heading & description prop 

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #4688 

## Summary

- Deprecated `itemTitle` props in favor of `heading` in  `accordion-item` and `stepper-item` components
- Deprecated `itemSubtitle` , `summary` in favor of `description` in `accordion-item` , `stepper-item`,`block` & `panel` components

### CODE REVIEW

File: src/components/accordion-item/accordion-item.tsx [hunk 1]
```diff
@@ -41,12 +41,28 @@ export class AccordionItem {
...
+
+  /**
+   * pass a title for the accordion item
+   *
+   * @deprecated use description instead
+   */
   @Prop() itemSubtitle?: string;
 
+  /**
+   * accordion item description
```
Comment (geospatialem) [inline src/components/accordion-item/accordion-item.tsx:62]: Recommended consistency update:
**The component description text.**

File: src/components/block/block.tsx [hunk 2]
```diff
@@ -99,9 +99,14 @@ export class Block implements ConditionalSlotComponent, InteractiveComponent {
 
   /**
    * Block summary.
+   *
+   * @deprecated use description instead
```
Comment (geospatialem) [inline src/components/block/block.tsx:103]: For consistency, could we change the `@deprecated` line to?:
**Use "description" for the component description text.**

File: src/components/block/block.tsx [hunk 3]
```diff
@@ -99,9 +99,14 @@ export class Block implements ConditionalSlotComponent, InteractiveComponent {
...
    */
   @Prop() summary: string;
 
+  /**Block description */
```
Comment (geospatialem) [inline src/components/block/block.tsx:107]: Recommended consistency update:
**The component description text.**

File: src/components/panel/panel.tsx [hunk 4]
```diff
@@ -117,9 +117,14 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
 
   /**
    * Summary text. A description displayed underneath the heading.
+   *
+   * @deprecated use description instead
    */
   @Prop() summary?: string;
 
+  /** Panel description */
```
Comment (geospatialem) [inline src/components/panel/panel.tsx:125]: Recommended consistency update:
**The component description text.**

File: src/components/stepper-item/stepper-item.tsx [hunk 5]
```diff
@@ -53,12 +53,26 @@ export class StepperItem implements InteractiveComponent {
...
   @Prop() itemTitle?: string;
 
-  /** pass a title for the stepper item */
+  /** stepper item heading */
+  @Prop() heading?: string;
+
+  /**
+   * pass a title for the stepper item
+   *
+   * @deprecated use description instead
```
Comment (geospatialem) [inline src/components/stepper-item/stepper-item.tsx:69]: For consistency, could we change the `@deprecated` line to?:
**Use "description" for the component description text.**

File: src/components/stepper-item/stepper-item.tsx [hunk 6]
```diff
@@ -53,12 +53,26 @@ export class StepperItem implements InteractiveComponent {
...
+   */
   @Prop() itemSubtitle?: string;
 
+  /** stepper item description */
```
Comment (geospatialem) [inline src/components/stepper-item/stepper-item.tsx:73]: Recommended consistency update:
**The component description text.**

**Review by geospatialem — COMMENTED** [review]
A few API consistency doc recommendations based on https://github.com/Esri/calcite-components/issues/4442

File: src/components/accordion-item/accordion-item.tsx [hunk 7]
```diff
@@ -41,12 +41,28 @@ export class AccordionItem {
...
   @Prop() itemTitle?: string;
 
-  /** pass a title for the accordion item */
+  /** accordion item heading */
+  @Prop() heading?: string;
+
+  /**
+   * pass a title for the accordion item
+   *
+   * @deprecated use description instead
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/accordion-item/accordion-item.tsx:57]: does that mean we don't use `@deprecated` tag at all?

[the remark this answers is not in the document]
Comment (geospatialem) [reply src/components/accordion-item/accordion-item.tsx:57]: Good question, sorry for the confusion. The text would be after the `@deprecated` tag.

[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/components/accordion-item/accordion-item.tsx:57]: Got you. Thanks.

**Pull request merged**